### ISSUE TITLE
Bug: [@typescript-eslint/no-unused-vars] False positive on export-equals syntax

### ISSUE THREAD

**Issue by Yoten** [typescript-eslint/typescript-eslint#6383]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=JYOwLgpgTgZghgYwgAgCoQM5gJLmvJZAbwFgAoZZGAe2oC5kspQBzAbnIF9yIAPAB2pQwyALxpMOPLEQQ2QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0URNGj5okbgF8QpoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEMCisJye6FYemArugDTg2R4sACacMbOvCFEAVmT40oQ0RgBK6ErFScCxCQDkRrBQMwatAN3QApTUQCyIsa0oduJ2hscr0ASSIsmLCIbG68-LR4AIaYAOboeBIaAKyQHlBEsACi8MzYRFmYQZihXOGKkJm+SKjI0MwAgkQAnmwAZlGoJOWm6EJWAMII8OgBZJQdXT20JBx1ePpiqAMAFujQANbj7GXpkJqcmNDo9lEhE53d-AC+INdAA

### Repro Code

```typescript
interface TestInterface {
  foo: string;
}
export = TestInterface;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "composite": true,
    "module": "commonjs",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "target": "es5",
    "noEmitOnError": true,
    "noImplicitAny": false,
    "removeComments": false,
    "strictNullChecks": true,
    "sourceMap": false
  }
}
```

### Expected Result

No errors given.

### Actual Result

"'TestInterface' is defined but never used."

### Additional Info

In the course of upgrading typescript-eslint from 3.9.1 --> 5.49.0 I started getting a ton of "no-used-var" failures from some of our older projects that still use the old "export = X", "import X = require(...)" syntaxes.

I'm guessing this happened with the scope analysis changes from 4.0? Regardless, I know it's a bit silly considering how out of date "export =" is, but I wasn't sure if they're supposed to be supported or not.

**Comment by bradzacher** [comment]
TBH I'm surprised this is actually allowed by TS.
I was under the impression that `export =` only allowed you to reference values because the idea behind them is that they define a commonjs `module.exports` export and that you use `import X = require('mod')` to import them.

Having a quick play - it looks like this works the same as if you had done `export default interface`.
I.e.
```ts
// foo.ts
interface Foo { prop: string }
export = Foo;

// bar.ts
import Foo = require('./foo');
Foo.prop; // error Foo is a type
type T = Foo['prop']; // correct
```

This is really weird - I've literally never seen anyone do this before - I didn't know that TS allowed this!

**Comment by Yoten** [comment]
> This is really weird - I've literally never seen anyone do this before - I didn't know that TS allowed this!

Yep, it's briefly mentioned in the docs: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

It's not exactly like using "export default Name" though. Using export-equals and import-require, you can then immediately use the interface as if it were a type, such as:

`var myArray: Array<TestInterface> = null;`

...and it works fine. However if you used "export default Name" the "import = require" will resolve successfully, but the above Array code will fail with a "cannot use namespace 'TestInterface' as a type" error. You'd have to use "typeof TestInterface" instead to make things work.
